geomstats/distributions/lognormal.py

```diff
@@ -0,0 +1,187 @@
+            shape=[n*(n+1)/2, n*(n+1)/2] if manifold is SPD Manifold
+        Covariance of the distribution.
+
+    Returns
+    --------
+    samples : array-like, shape=[..., dim]
+        Samples from LogNormal distribution.
+
+    Example
+    --------
+    >>> import geomstats.backend as gs
+    >>> from geomstats.geometry.spd_matrices import SPDMatrices
+    >>> from geomstats.distributions.lognormal import LogNormal
+    >>> mean = 2 * gs.eye(3)
+    >>> cov  = gs.eye(6)
+    >>> SPDManifold = SPDMatrices(3)
+    >>> LogNormalSampler = LogNormal(SPDManifold, mean, cov)
+    >>> data = LogNormalSampler.sample(5)
+
+    References
+    ----------
+    .. [LNGASPD2016] A. Schwartzman,
+    "LogNormal distributions and"
+    "Geometric Averages of Symmetric Positive Definite Matrices.",
+    International Statistical Review 84.3 (2016): 456-486.
+    """
+    def __init__(self, manifold, mean, cov=None):
+
+        if (
+            not isinstance(manifold, SPDMatrices) and
+            not isinstance(manifold, Euclidean)
+        ):
+            raise ValueError(
+                "Invalid Manifold object, "
+                "Should be of type SPDMatrices or Euclidean")
+
+        if not manifold.belongs(mean):
+            raise ValueError(
+                "Invalid Value in mean, doesn't belong to ",
+                type(manifold).__name__)
+
+        n = mean.shape[-1]
+        metric = manifold.metric
+        if isinstance(manifold, Euclidean):
+            cov_n = n
+            if metric is None:
+                manifold.metric = EuclideanMetric(n)
+            else:
+                metric = manifold.metric
+                if not isinstance(metric, EuclideanMetric):
+                    raise ValueError(
+                        "Invalid Metric, "
+                        "Should be of type EuclideanMetric")
+
+        else:
+            cov_n = (n * (n + 1)) // 2
+            if metric is None:
+                manifold.metric = SPDMetricLogEuclidean(n)
+            else:
+                metric = manifold.metric
+                if (
+                    not isinstance(metric, SPDMetricLogEuclidean) and
+                    not isinstance(metric, SPDMetricAffine)
+                ):
+                    raise ValueError(
+                        "Invalid Metric, "
+                        "Should be of type SPDMetricLogEuclidean"
+                        "or SPDMetricAffine")
+
+        if cov is not None:
+            if (
+                cov.ndim != 2 or
+                (cov.shape[0], cov.shape[1]) != (cov_n, cov_n)
+            ):
+                valid_shape = (cov_n, cov_n)
+                raise ValueError("Invalid Shape, "
+                                 "cov should have shape", valid_shape)
+
+        else:
+            cov = gs.eye(cov_n)
+
+        self.__manifold = manifold
+        self.__mean = mean
+        self.__cov = cov
+        self.__mean_dim = n
```

Comment:
maybe `__mean_shape` which would be `n` or `(n, n)`? dim could be slightly confusing.

Reply:
or `__point_shape` because this is, more generally, the shape of the points on that manifold.